sling/nlp/wiki/wiki-annotator.cc

```diff
@@ -273,8 +273,18 @@ void WikiAnnotator::Category(const Node &node,
 void WikiAnnotator::AddToDocument(Document *document) {
   // Add annotated spans to document.
   for (Annotation &a : annotations_) {
-    int begin = document->Locate(a.begin.AsInt());
-    int end = document->Locate(a.end.AsInt());
+    // Get character span for annotation.
+    int b = a.begin.AsInt();
+    int e = a.end.AsInt();
+
+    // Map character span to token span. Discard annotation if span is not
+    // aligned with tokens.
+    int begin = document->Locate(b);
+    int end = document->Locate(e);
+    if (begin == end) continue;
+    if (document->token(begin).begin() != b) continue;
```

Comment:
Just a comment. Not sure how to fix this easily. Maybe if it's just me, but when you're not super familiar with the code it can be hard to tell token number/index ints from document/string/text positions/index ints.

Reply:
I tried to make it clear in the comment that we are mapping from character to token positions here. Luckily, we mostly deal with token positions in the remaining parts of the code.